File: 2025-01-29-mars-rover-py/rover.py


def rover(position_commands):
    y = position_commands["y"]
    x = position_commands["x"]
    direction = position_commands["direction"]
    for command in position_commands['commands']:
        if command == 'f':
            x += 1
    return {"x": x, "y": y, "direction": direction}

File: 2025-09-16-robot-and-react/src/lib/Rover.ts
type Coordinates = {
  x: number;
  y: number;
  direction: string;
};

type Command = "R";

const DIRECTIONS = ["N", "E", "S", "W"];

const Rover = ({
  coordinates,
  commands = [],
}: {
  coordinates: Coordinates;
  commands?: Command[];
}) => {
  const { x, y } = coordinates;
  let { direction } = coordinates;

  commands.forEach((command) => {
    if (command === "R") {
      const index = DIRECTIONS.findIndex((dir) => dir === direction);
      direction = DIRECTIONS[(index + 1) % DIRECTIONS.length];
    }
  });
  return { x, y, direction };
};

export default Rover;
